electrumsv/wallet_database/migrations/migration_0029_reference_server.py

```diff
@@ -166,6 +168,28 @@ def execute(conn: sqlite3.Connection, password_token: PasswordTokenProtocol,
     # they may not be re-verified until they jump through server hoops.
     conn.execute("UPDATE Transactions SET proof_data=NULL")
 
+    # Transfer all merkle proof data from the Transactions table to the
+    # The pathways for insertion to this table are as follows:
+    #  1) Wallet._obtain_merkle_proofs_worker_async -> Wallet.import_transaction_async
+    #  2) Wallet._obtain_transactions_worker_async -> Wallet.import_transaction_async
+    #  3) wait_for_merkle_proofs_and_double_spends (not yet in use) - mAPI callbacks
+    # All proofs from all chains should be inserted here (i.e. including orphaned proofs). They
+    # can be pruned when the proof on the main server chain is buried by sufficient proof of work .
+    conn.execute("""CREATE TABLE IF NOT EXISTS TransactionProofs (
+        block_hash BLOB,
+        tx_hash BLOB,
+        proof_data BLOB DEFAULT NULL,
+        block_height INTEGER DEFAULT NULL,
+        block_position INTEGER DEFAULT NULL
+    )""")
+    conn.execute("CREATE INDEX idx_tx_proofs_block_hash ON TransactionProofs (block_hash)")
```

Comment:
We only do unique indexes, what are these and why are they here?

Reply:
There can be more than one proof per transaction hash and more than one transaction per block. 

Best I can do is a clustered index of (tx_hash, block_hash)?

Reply:
What I mean is these are optimisation indexes and not unique indexes? We do not do optimisation indexes until it has been shown to be necessary.